pkgs/native_assets_cli/lib/src/model/build_config.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:cli_config/cli_config.dart';
+import 'package:collection/collection.dart';
+
+import '../utils/map.dart';
+import '../utils/yaml.dart';
+import 'ios_sdk.dart';
+import 'metadata.dart';
+import 'packaging_preference.dart';
+import 'target.dart';
+
+class BuildConfig {
+  /// The folder in which all output and intermediate artifacts should be
+  /// placed.
+  Uri get outDir => _outDir;
+  late final Uri _outDir;
+
+  /// The root of the package the native assets are built for.
+  ///
+  /// Often a package's native assets are built because a package is a
+  /// dependency of another. For this it is convenient to know the packageRoot.
+  Uri get packageRoot => _packageRoot;
+  late final Uri _packageRoot;
+
+  /// The target that is being compiled for.
+  Target get target => _target;
+  late final Target _target;
+
+  /// When compiling for iOS, whether to target device or simulator.
+  ///
+  /// Required when [target.os] equals [OS.iOS].
+  IOSSdk? get targetIOSSdk => _targetIOSSdk;
+  late final IOSSdk? _targetIOSSdk;
+
+  /// Path to a C compiler.
+  Uri? get cc => _cc;
```

Comment:
Why does the native_assets_cli have C specific stuff here? I thought it was language independent and delegated all that stuff to c_compiler? What happens when we add Rust etc?

Reply:
It's optional. We fall back to detecting a C compiler on the host machine if it's not supplied.
The reason it's part of the API here is that Flutter already selects a C compiler (with the logic visible in `flutter doctor`). And we should build the C code with the same compiler.

For rust, as long as we don't have a Rust compiler in our own dependencies, we will try to just locate it on the host machine.
If we ever start requiring a Rust compiler for Flutter app builds, we can add an optional `rustc` Uri in this config.